### PR TITLE
feat: add new linkdrop standard fixes & trial account offboarding

### PR DESCRIPTION
**Pull request by BenKurrek**
## Changes description

- Layed the foundational work for different linkdrop types and rendering the linkdrop landing page accordingly.
- Introduced Trial Account Offboarding flow as a new type of linkdrop (more info found [here](https://docs.keypom.xyz/docs/next/TrialAccounts/offboarding))
- New linkdrop standard calls for different interfaces and methods. This has been (mostly) addressed in this PR
- Small bugfixes

## Demo

Trial Account flow is as follows:
1. The linkdrop landing page now looks as follows (this is *only* for trial accounts):
image
2. When the button is clicked, they go through the regular linkdrop claiming flow (except they don't choose an account ID):

- Choose recovery method

- Verify recovery method

- Create Account

A full demo:
https://drive.google.com/file/d/1hD1oFvO_RG5-PvzWJUvj-B96rniWY2zO/view?usp=share_link

### CODE REVIEW

File: packages/frontend/src/utils/wallet.js [hunk 1]
```diff
@@ -616,7 +636,7 @@ export default class Wallet {
     await contract.create_account_and_claim(
       {
         new_account_id: accountId,
-        new_public_key: publicKey.toString().replace("ed25519:", ""),
+        new_public_key: publicKey.toString()//.replace("ed25519:", ""),
```
Comment (lostpebble) [inline packages/frontend/src/utils/wallet.js:639]: Please remove instead of comment if not relevant anymore

Comment (BenKurrek) [reply packages/frontend/src/utils/wallet.js:639]: Oh sorry I missed this - I'm going to keep it, it was just breaking something for my trial contract but this has been fixed.

**Comment by BenKurrek** [comment]
# Testing

## Keypom Trial Account (no wallet logged in yet)

I've tested claiming a trial account to a fresh instance of MNW (no prior state or accounts logged in yet).
Here is the demo: https://drive.google.com/file/d/1J7vxQYhEDMsjbpPeWNlkUwdLaAgXm22P/view?usp=share_link

Links for you try yourselves:
https://localhost:1234/linkdrop/trial-account-1681927518185.testnet/2dkgUefC2QPwSWupGPGQkmtyaaLzvpw3dch9w6zd8XGKPwjqtbqCoy58N1CAfuyojmYme9momPgmM1Jq6NxLz2W8
https://localhost:1234/linkdrop/trial-account-1681927549042.testnet/4ADYk3Tironn55y9o48LoMbbvfYEBehobNWK4xH9i8YWMXj976NNM8Y4DszTeB7nCQoLZcYBu7R3hn1jP4R47L8K
https://localhost:1234/linkdrop/trial-account-1681927576027.testnet/39hvzyY2DXo5rV28Hp2A7WztfZdbRqYgEEEZg8Sx6sB1F9mRpE8uDfXg9eUdvd2zobYj2vwxJY5U7YuxBVw7A7AS

## Keypom Trial Account (wallets logged in already)

I've tested claiming a trial account in an instance of MNW that has wallets signed in already
Here is the demo: https://drive.google.com/file/d/1wLuU9N_8qtDLDcWELBAXeRL0iABaZd6Y/view?usp=share_link

Links for you to try:
https://localhost:1234/linkdrop/trial-account-1681928376420.testnet/3b9Rc46rZ4pxgq42vUmL1vBGsKoPHZrAXPgXihM4FbAYUYZJnN4Zu2gg97A6VvRanL54vEQpm5RxnDasD9GHnmfL
https://localhost:1234/linkdrop/trial-account-1681928401747.testnet/3UKAcMToJt9Q1pXts32S1fGMuTMg81DJvKTFFs5eGPdFHfHzm9SUiVxfpJcDc8SaBUMzzB55vXpNLo6PyXCM79A7
https://localhost:1234/linkdrop/trial-account-1681928437632.testnet/2SFm9FCQScwkjmCLqoJGMZ1rx5t5eouBp5RyXUKnKLHEaaXphZdxF5iySMD6nECWqDibJdLDE6hxVnCjnqSkbpxn

## Regular Linkdrop from Keypom Claiming to a New Account (no wallets logged in)
I've tested claiming a regular linkdrop to a new account coming from keypom in an instance of MNW that has NO wallets signed in:
Here is the demo: https://drive.google.com/file/d/1fTg9XyWhNCkkrCO21FIxHYrzwGTCypNI/view?usp=share_link

Links for you to try:
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:4AvWG5v1LkPkNFhEpxyBquwo5V3t2GcWgYM3xZ9KfSiXiq9aVTTpbJG3baB4MtC9GTve3zwo1pAycGDmoXZGV7z2',
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:ze3WP4A6wJyWHZo8nFuAeVdDJDNdNYNW9yzh7ZjGy5433D5uA7LNfeWC4zJuvuuKE3GLtCsaqJRekGxZCbgN5Lt',
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:3QRVjFYveCtpZ1PSsRFZYA5tKWHTTCihTnD4X2RUZnxxa2HLRvinQvH2h2Z3GmhPbBDLTRZ2n3wSx4kArTt5JjMx'

## Regular Linkdrop from Keypom Claiming to a New Account (wallets logged in already)
I've tested claiming a regular linkdrop to a new account coming from keypom in an instance of MNW that has wallets signed in already
Here is the demo: https://drive.google.com/file/d/1V5NXs4Lk_WhL-0wKWbS7IbY0GmAjHf0c/view?usp=share_link

Links for you to try:
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:4FufpZCkg9wKAfuzcYbUd2qUQbfrW1C7RkQWGRceEC45x53A6WV639pQtLxVetqztdidGw3kwpLpjpDwVb16qgXR',
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:2fjKbcN5pDJ6yQMggCFJnP8yzy8Exfx1TDjPWbBKnN9DokXZDR9VBxf9jAmQ8P63Jjc65BpGrqYkvQZ3vecxjTLE',
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:UGhi8epFGKpjGPfVjxBsnRGCrvFBSFu5TrSGHXqXgQqqc5p51eLSab5oxfz7LbBoo7wwy4V5CXPzup6E6bS2ttD'

## Regular Linkdrop from Keypom Claiming to an Existing Account (wallets logged in already)
I've tested claiming a regular linkdrop to an existing account coming from keypom in an instance of MNW that has wallets signed in already
Here is the demo: https://drive.google.com/file/d/1R7uZp89omXO0Oi2UnWWvVao40KvnBeZA/view?usp=share_link

Links for you to try:
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:2zqBGgB3FzKgNA1Ha4AQ3eojy4NyFj2emD1fDmaWXa1m2qBggaY9Ew7tHgQ3vbBqetcCDnAH9xU1gk9AHpabXWxY',
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:45AGNkVvq35aUvUKpbdH2bwFyWtPnVpdcPe8DHVe17AXtEAH8VDUBpWL8adXjK9pz9zDwac9NymomoFPLqMLR3q6',
'https://localhost:1234/linkdrop/v2.keypom.testnet/ed25519:4ZnwWGUGLZF79adyU1kVEQsQTeqRGiX9o3d2Q4Mq6M8nganmEq2c6eRSPqFQy2JeCQ21b5eahe4nDNUvT6tV8PNE'

## Regular Linkdrop Create Account (no wallet logged in yet)
Tested claiming a regular linkdrop to a new account coming from the canonical `near` linkdrop contract (not Keypom). MNW had no wallets signed in.
Demo: https://drive.google.com/file/d/14XHBx0Sh_AVQtdzViM4-bdxp1IzGis7P/view?usp=share_link

Links for you to try:
'https://localhost:1234/linkdrop/testnet/2eZCCTPyo9TJUhdH5EEVC5mynyJqjEcs8Z4JemEXo5hbLe8Pru4Lz4BrvFFsgVJBiiaVx97eyrFR7DztVqAhgifS',
'https://localhost:1234/linkdrop/testnet/4GCuWz3HGEpChFLcpytSLX5hbipHNV9WwV7Ac1XEdAKUb4u523uWpkPj5kZQr5vp6auaamNDeV1jR172AnWbs1sk',
'https://localhost:1234/linkdrop/testnet/4CrWH7MYgZBJZSDYJDb1vHgT3ukWU1WK6bnphKuSMp749SwzKCH276G4XSweRh3RMMBPBqUoky3cji5DXJF98Cqv'

## Regular Linkdrop Create Account (wallets logged in)
Tested claiming a regular linkdrop to a new account coming from the canonical `near` linkdrop contract (not Keypom). MNW had wallets signed in already.
Demo: https://drive.google.com/file/d/1FhozIixjuBBnh0t1su4sP3FctKo0nFSk/view?usp=share_link

Links for you to try:
'https://localhost:1234/linkdrop/testnet/3VLUKztpVTPeTRKDLXN7eY8MjVx9rSB8QCYACwGeGq8JCH8GmW4uo3qUUeh9Z7RD9LJCGv4W36bKQaQxwfw7KbGq',
'https://localhost:1234/linkdrop/testnet/2itUibZkSLJ8GrN7JDeGj1naa276UQPA3Q2Zyk5wUcBAuG6UhHQKuo6tAunsxHDVtaAM29zjGLwMX2A1JysdGJBX',
'https://localhost:1234/linkdrop/testnet/3FwoUHMC9nSfwHymuEx9qzPTxkrYgd2zpnqPv9HWy7xVCA9UL5qMXd3SrUhKJPCnqYAcfQaWc7vW98GPoCf8Tkb3'

## Regular Linkdrop Claim to Existing Account
Tested claiming a regular linkdrop to an existing account coming from the canonical `near` linkdrop contract (not Keypom).
Demo: https://drive.google.com/file/d/1Po_Zi0vNeZ0RJW5XtaRXJiAgAfiguO_G/view?usp=share_link

Links for you to try:
'https://localhost:1234/linkdrop/testnet/5CE76NxmH7i3DyAiscjFBWxF82AUUx8hAoJUQLNwFoAyWT3oaxtFdFyY2JorCVM4XbvAttk6pFyXW4tZjmKdjQ1A',
'https://localhost:1234/linkdrop/testnet/4WJC7dDwZtmPDRdwFECzj2iL3cMM9dUkrtW52JHV1ibtDbqPD4T1CPvJBPoP5mTiYVr2P9vANdWA4KVQQYZH7WDt',
'https://localhost:1234/linkdrop/testnet/9u7WccJJBrkPjT8HDA5rgWsfCXvvMZQdnrfsH53YPZHPzBDrLCgdZm1qkXTLBjvE1oZG2dYNTXWvmnun4rf8kMo'

**Comment by DominicGBauer** [comment]
@lostpebble can we close this now since it's been merged in?

**Pull request closed without merging**